doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,449 @@
+
+```py
+>>> from sympy import solveset, S
+>>> from sympy.abc import x
+>>> solution = solveset(x**2 + 1, x, domain=S.Reals)
+>>> print(solution)
+EmptySet
+```
+
+### Explicitly represent infinite sets of possible solutions using {func}`~.solveset`
+
+{func}`~.solveset` 
+[can represent infinite sets of possible solutions](why-solveset)
+and express them in standard mathematical notation, 
+for example $\sin(x) = 0$ for $x = n * \pi$ for every integer value of $n$:
+
+```py
+>>> from sympy import pprint, sin, solveset
+>>> from sympy.abc import x
+>>> solution = solveset(sin(x), x)
+>>> pprint(solution)
+{2*n*pi | n in Integers} U {2*n*pi + pi | n in Integers}
+```
+
+However, {func}`~.solve` will return only a finite number of solutions:
+
+```py
+>>> from sympy import sin, solve
+>>> from sympy.calculus.util import periodicity
+>>> from sympy.abc import x
+>>> f = sin(x)
+>>> solution = solve(f, x)
+>>> print(solution)
+[0, pi]
+>>> print(periodicity(f, x))
+2*pi
+```
+
+{func}`~.solve` tries to return just enough solutions so that all (infinitely many) solutions can generated 
+from the returned solutions by adding integer multiples of the {func}`~.periodicity` of the equation, here $2\pi$.
+
+## Use the solution result
+
+### Substitute solutions from {func}`~.solve` into an expression
+
+You can substitute solutions from {func}`~.solve` into an expression.
+
+A common use case is finding the critical points and values for a function $f$. 
+At the critical points, the {class}`~.Derivative` equals zero (or is undefined). 
+You can then obtain the function values at those critical points
+by substituting the critical points back into the function using {meth}`~sympy.core.basic.Basic.subs`. You can also tell if the critical point is a maxima or minima by substituting the values
+into the expression for the second derivative: a negative value indicates a maximum, and a positive value indicates a minimum.
+
+```py
+>>> from sympy.abc import x
+>>> from sympy import solve, diff
+>>> f = x**3 + x**2 - x
+>>> derivative = diff(f, x)
+>>> critical_points = solve(derivative, x, dict=True)
+>>> print(critical_points)
+[{x: -1}, {x: 1/3}]
+>>> point1, point2 = critical_points
+>>> print(f.subs(point1))
+1
+>>> print(f.subs(point2))
+-5/27
+>>> curvature = diff(f, x, 2)
+>>> print(curvature.subs(point1))
+-4
+>>> print(curvature.subs(point2))
+4
+```
+
+### {func}`~.solveset` solution sets cannot necessarily be interrogated programmatically
+
+If {func}`~.solveset` returns a finite set (class {class}`~.FiniteSet`), you can iterate through the solutions:
+
+```py
+>>> from sympy import solveset
+>>> from sympy.abc import x, y
+>>> solution_set = solveset(x**2 - y, x)
+>>> print(solution_set)
+{-sqrt(y), sqrt(y)}
+>>> solution_list = list(solution_set)
+>>> print(solution_list)
+[sqrt(y), -sqrt(y)]
+```
+
+However, for more complex results, it may not be possible to list the solutions:
```

Comment:
This should explain both why `solveset` returned an unevaluated `Intersection` and also how to avoid this:

- Declare `y` to be real and positive
- Extract the FiniteSet from the Intersection.

Reply:
>why solveset returned an unevaluated Intersection

>Extract the FiniteSet from the Intersection

These would be good to include, and are beyond my SymPy knowledge, nor could I find them in the documentation. Could you point me to, or give, an explanation/example as a basis for me to document?

Reply:
I worked up an example that I believe addresses this suggestion.

Reply:
To get the FiniteSet from an intersection you can use `.args`.